src/php/globals.rs

```diff
@@ -16,8 +16,32 @@ impl ExecutorGlobals {
             .expect("Static executor globals were invalid")
     }
 
+    fn get_mut() -> &'static mut Self {
+        // SAFETY: PHP executor globals are statically declared therefore should never
+        // return an invalid pointer.
+        // TODO: Should this be syncronized?
+        unsafe { ext_php_rs_executor_globals().as_mut() }
+            .expect("Static executor globals were invalid")
+    }
+
     /// Attempts to retrieve the global class hash table.
     pub fn class_table(&self) -> Option<&HashTable> {
         unsafe { self.class_table.as_ref() }
     }
+
+    /// Attempts to extract the last PHP exception captured by the interpreter.
+    ///
+    /// Note that the caller is responsible for freeing the memory here or it'll leak.
+    pub fn take_exception() -> Option<*mut ZendObject> {
```

Comment:
Return `*mut ZendObject` or load the pointer into a `Zval`, returning that? Or potentially `&'static ZendObject` (if it is non-null and well aligned)?

Reply:
Returning a `Zval` just gives it a nice wrapper which will drop the object when `Zval` goes out of scope.

Reply:
This is fine for now since it's not getting used anywhere, but eventually should be contained in a `ZBox`.